proving-system/stark/src/prover.rs

```diff
@@ -0,0 +1,237 @@
+use super::constraints::boundary::{BoundaryConstraint, BoundaryConstraints};
+use super::fri::fri_decommit::fri_decommit_layers;
+use super::utils::compute_zerofier;
+use super::{
+    generate_primitive_root, generate_roots_of_unity_coset, StarkQueryProof, COSET_OFFSET, FE,
+    ORDER_OF_ROOTS_OF_UNITY_FOR_LDE, ORDER_OF_ROOTS_OF_UNITY_TRACE,
+};
+use lambdaworks_crypto::fiat_shamir::transcript::Transcript;
+use lambdaworks_math::polynomial::{self, Polynomial};
+use std::ops::{Div, Mul};
+
+pub fn prove(trace: &[FE]) -> StarkQueryProof {
+    let transcript = &mut Transcript::new();
+
+    // * Generate Coset
+    let trace_primitive_root = generate_primitive_root(ORDER_OF_ROOTS_OF_UNITY_TRACE);
```

Comment:
this has been changed, you need to sync with main

Reply:
We are pointing this PR to `deep-fri` branch, so I think we should hange this when we want to merge that branch to main.

Reply:
Ah sorry, I missed that. Thanks!